src/context/Confirmation.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+    Button,
+    Dialog,
+    DialogActions,
+    DialogContent,
+    DialogContentText,
+    DialogTitle,
+} from '@mui/material';
+import React, { createContext, useContext, useRef, useState } from 'react';
+import { FormattedMessage } from 'react-intl';
+
+export interface IConfirmationModalOptions {
+    confirmText?: string;
+    cancelText?: string;
+    title?: string;
+    message: string;
+}
```

Comment:
All things are option except for the main message that is getting displayed.

Reply:
I find interfaces with optional properties easier to understand when the required properties are listed before those that are optional. More often than not, required properties are of primary concern (as without them, the object cannot be created); thus, as the file is read from top to bottom, having them be encountered before the optional properties is a preference of mine. Just sharing my two cents on the matter.